tests/src/org.testcontainers/testcontainers/1.19.8/src/main/java/org_testcontainers/testcontainers/FindJsonSerialization.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_testcontainers.testcontainers;
+
+import com.github.dockerjava.api.model.DockerObject;
+import org.reflections.Reflections;
+import org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Modifier;
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+
+import static org.reflections.scanners.Scanners.SubTypes;
+
+class FindJsonSerialization {
```

Comment:
Why did you need this way of collecting metadata? Couldn't the agent collect this metadata on a proper test?

Reply:
I just re-applied what they did in the tests for the previous versions. 

My interpretation is the following: the agent does not catch any of those APIs, because the tests do not exercise these classes. The testcontainers API, and the underlying docker APIs, is huge. For example, the core testcontainers configuration interface, `Container`, has around 60 methods. It's most common implementation has 114 public methods. The methods, in turn, often use their own types (`Network`, `DockerClient`, ...). While there is redundancy in those methods, it's still a lot of surface to cover in tests. So, the initial implementation was done with a rather scattershot method that brings in the full - probably too much, but this being **test**containers, maybe the thinking is "test binaries are not as critical as production binaries" ?

To do this properly, we would have to spend quite a lot of time to nail down the correct tests, and maybe bring the Testcontainers team on board. I'm happy on refining the tests later on, for a later version, but I don't have the bandwidth right now.